codex/node.nim

```diff
@@ -571,25 +584,28 @@ proc onProve(
 
   trace "Received proof challenge"
 
-  without cid =? Cid.init(cidStr).mapFailure, err:
-    error "Unable to parse Cid", cid, err = err.msg
-    return failure(err)
+  if prover =? self.prover:
+    trace "Prover enabled"
```

Comment:
Maybe easier to read by returning a failure if `self.prover.isNone`:
```suggestion
  without prover =? self.prover:
    return failure "no prover"
```

Reply:
Yes, but I wont change it for now, because it will break the integration tests